test/integration/tcp_tunneling_integration_test.cc

```diff
@@ -311,11 +311,12 @@ INSTANTIATE_TEST_SUITE_P(IpVersions, ConnectTerminationIntegrationTest,
                          testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
                          TestUtility::ipTestParamsToString);
 
-// Tunneling downstream TCP over an upstream HTTP channel.
-class TcpTunnelingIntegrationTest : public testing::TestWithParam<Network::Address::IpVersion>,
-                                    public HttpIntegrationTest {
+// Tunneling downstream TCP over an upstream HTTP/2 CONNECT tunnel.
+class H2TcpTunnelingIntegrationTest : public testing::TestWithParam<Network::Address::IpVersion>,
```

Comment:
Can we not just add a param of codec to this test?
I'd prefer we have all the same tests run for HTTP/1 and HTTP/2, modulo early return for any tests which have multiple streams for HTTP/1

I think we basically want all the same tests for HTTP/2 running for HTTP/1, and then we want a few "raw TCP" tests for HTTP/1 which test various content length and transfer encoding issues.

Reply:
Yep, I also thought that it's better to normalize it, I'll give it a try